### ISSUE TITLE
Shape mismatch across ESMProteinTensor and forward pass for ESM3 

### ISSUE THREAD

**Issue by devpatelio** [evolutionaryscale/esm#56]
Hey ESM Team! I'm not sure if I'm passing in my raw sequences correctly for the tracks, but when it comes to extracting the logits through a simple forward pass, I think there is a shape error where the tokenized inputs have the incorrect shape. You have to manually reassign the Protein Tensor instance with a tensor that has an additional dimension. Without the unsqueeze line, it throws a tuple index out of range error. 
 
```
from esm.models.esm3 import *
fconf = ForwardConfig(return_logits=ReturnLogitsConfig(sequence=True))
print(fconf.return_logits)

# gconf = SamplingConfig()
inp = 'MLPGLALLLLAAWTARALEVPTDGNAGLLAEPQIAMFCGRLNMHMNV'
def esm3_forward(seq, client, config=fconf):
    protein = ESMProtein(sequence=seq)
    prot_obj = client.encode(protein)
    # prot_obj.sequence = torch.unsqueeze(prot_obj.sequence, 0)
    print(prot_obj.sequence.shape)

    output = client._forward(
        input = prot_obj,
        config = config
    )

    return output
```

**Comment by elhambuxton** [comment]
I had the same problem. Is it because we are doing a forward pass on a single sequence and missing batch dimension?

**Comment by santiag0m** [comment]
You should avoid calling the `client._forward` method directly whenever you can, as it does require extra batching logic that the library handles internally.

For most use cases, `client.forward_and_sample` should be preferred.

**Comment by ebetica** [comment]
@santiag0m I do think this is a bug though, we should try to fix it for `._forward`.

**Comment by santiag0m** [comment]
It is related to https://github.com/evolutionaryscale/esm/issues/67. Will look into a nicer way to get logits out.

**Comment by ebetica** [comment]
We introduced the `.logits` function to replace `._forward`. I think should satisfy the concerns here? Please open a new issue if that doesn't work for you.
